Yank/repex.py

```diff
@@ -70,26 +70,50 @@ class Reporter(object):
 
     Parameters
     ----------
-    storage : str
-        The path to the file. In the future this will be able to take Storage
-        classes as well.
+    base_storage : str
+        The path to the files base name, this should not include a file suffix
+        The multiple storage files will be saved as two files named
+            "base_storage + _checkpointing" and "base_storage + _analysis"
+        In the future this will be able to take Storage classes as well.
     open_mode : str or None
         The mode of the file between 'r', 'w', and 'a' (or equivalently 'r+').
         If None, the storage file won't be open on construction, and a call to
         Reporter.open() will be needed before attempting read/write operations.
+    checkpoint_interval : int >= 1, Default: 10
+        The frequency at which checkpointing information is written relative to analysis information. This is a multiple
+        of the iteration at which energies is written, hence why it must be greater than or equal to 1.
+        Checkpoint information cannot be written on iterations which where iteration % checkpoint_interval != 0.
+        Attempting to read checkpointing information results in a masked array where only entries which were written
+            are unmasked
 
     """
-    def __init__(self, storage, open_mode=None):
-        self._storage_file_path = storage
-        self._storage = None
+    def __init__(self, base_storage, open_mode=None, checkpoint_interval=10):
```

Comment:
I would make the constructor signature something like
```python
def __init__(self, storage, open_mode=None, checkpoint_interval=10, checkpoint_storage=None):
```
instead, and if `checkpoint_storage is None`, it will take the `storage` path and add a `_checkpoint` suffix to the name (plus append the correct extension). This way:
- We'd be able to pass a `Storage` objects to the constructor in the future in alternative to paths.
- We'd be able in the future to store positions and all the other informations in different formats by passing two different `Storage` objects or file paths with different extensions.

If a user wants to have a different name for the checkpoint than the default (probably a rare use case), he would pass directly a `Reporter` object instead of a file path. Something like:
```python
reporter = Reporter(storage='complex.nc', checkpoint_interval=50, checkpoint_storage='some_other_name.nc')
alchemical_phase.create(..., storage=reporter)
```